Assets/SteamVR_Unity_Toolkit/Scripts/VRTK_PlayerClimb.cs

```diff
@@ -0,0 +1,201 @@
+﻿namespace VRTK
+{
+    using UnityEngine;
+    using System.Collections;
+
+    public struct PlayerClimbEventArgs
+    {
+        public uint controllerIndex;
+        public GameObject target;
+    }
+
+    public delegate void PlayerClimbEventHandler(object sender, PlayerClimbEventArgs e);
+
+    public class VRTK_PlayerClimb : MonoBehaviour
+    {
+        public event PlayerClimbEventHandler PlayerClimbStarted;
+        public event PlayerClimbEventHandler PlayerClimbEnded;
+
+        public bool usePlayerScale = true;
+        public bool useGravity = true;
+
+        private Transform controllerTransform;
+        private Vector3 startControllerPosition;
+        private Vector3 startPosition;
+
+        private Vector3 lastGoodHeadsetPosition;
+        private bool headsetColliding = false;
+        private bool isClimbing = false;
+
+        private VRTK_PlayerPresence playerPresence;
+        private bool lastGravitySetting;
+
+        private virtual void OnPlayerClimbStarted(PlayerClimbEventArgs e)
+        {
+            if (PlayerClimbStarted != null)
+            {
+                PlayerClimbStarted(this, e);
+            }
+        }
+
+        private virtual void OnPlayerClimbEnded(PlayerClimbEventArgs e)
+        {
+            if (PlayerClimbEnded != null)
+            {
+                PlayerClimbEnded(this, e);
+            }
+        }
+
+        private PlayerClimbEventArgs SetPlayerClimbEvent(uint controllerIndex, GameObject target)
```

Comment:
This payload needs documenting in the `Class Events` bit of the docs. Have a look at:

https://github.com/mattboy64/SteamVR_Unity_Toolkit/blob/7ae25efa8cc081f19ae67d8b8087e81673e359f0/DOCUMENTATION.md#event-payload-5

To see an example of how to describe the payload